pkg/component/controller/cplb_reconciler.go

```diff
@@ -72,23 +72,21 @@ func (r *CPLBReconciler) Stop() {
 
 func (r *CPLBReconciler) watchAPIServers(ctx context.Context, clientSet kubernetes.Interface) {
 	for {
-		select {
-		default:
-			err := watch.Endpoints(clientSet.CoreV1().Endpoints("default")).
-				WithObjectName("kubernetes").
-				Until(ctx, func(endpoints *corev1.Endpoints) (bool, error) {
-					r.maybeUpdateIPs(endpoints)
-					return false, nil
-				})
-			// Log any reconciliation errors, but only if they don't
-			// indicate that the reconciler has been stopped.
-			if err != nil && !errors.Is(err, ctx.Err()) {
-				r.log.WithError(err).Error("Failed to reconcile API server addresses")
-			}
-
-			// After a watch error wait 5 seconds before retrying
-			time.Sleep(5 * time.Second)
+		err := watch.Endpoints(clientSet.CoreV1().Endpoints("default")).
+			WithObjectName("kubernetes").
+			Until(ctx, func(endpoints *corev1.Endpoints) (bool, error) {
+				r.maybeUpdateIPs(endpoints)
+				return false, nil
+			})
+		// Log any reconciliation errors, but only if they don't
+		// indicate that the reconciler has been stopped.
+		if err != nil && !errors.Is(err, ctx.Err()) {
+			r.log.WithError(err).Error("Failed to reconcile API server addresses")
+		}
 
+		// After a watch error wait 5 seconds before retrying
+		select {
+		case <-time.After(5 * time.Second):
```

Comment:
Couldn't we simplify this if we'd use `WithErrorCallback` and return 5sec duration from the callback?

Reply:
Indeed!